tests/testthat/test-tm_g_bivariate.R

```diff
@@ -0,0 +1,364 @@
+testthat::test_that("tm_g_bivariate creates a `teal_module` object", {
+  local_logger_threshold(logger::FATAL) # Suppress logger messages
+
+  testthat::expect_s3_class(
+    tm_g_bivariate(
+      "a label",
+      mock_data_extract_spec(select_multiple = FALSE),
+      mock_data_extract_spec(select_multiple = FALSE),
+      row_facet = mock_data_extract_spec(select_multiple = FALSE),
+      col_facet = mock_data_extract_spec(select_multiple = FALSE),
+      facet = TRUE,
+      color_setting = TRUE,
+      use_density = TRUE,
+      free_x_scales = TRUE,
+      free_y_scales = TRUE,
+      plot_height = c(400, 100, 600),
+      plot_width = c(600, 100, 600),
+      rotate_xaxis_labels = TRUE,
+      swap_axes = TRUE
+    ),
+    "teal_module"
+  )
+})
+
+testthat::test_that("tm_g_bivariate creates a `teal_module` object with default options", {
+  local_logger_threshold(logger::FATAL) # Suppress logger messages
+
+  testthat::expect_s3_class(
+    tm_g_bivariate(
+      "a label",
+      mock_data_extract_spec(select_multiple = FALSE),
+      mock_data_extract_spec(select_multiple = FALSE)
+    ),
+    "teal_module"
+  )
+})
+
+testthat::test_that("tm_g_bivariate creates a `teal_module` object with multiple data extract specs", {
+  local_logger_threshold(logger::FATAL) # Suppress logger messages
+
+  testthat::expect_s3_class(
+    tm_g_bivariate(
+      "a label",
+      list(mock_data_extract_spec(select_multiple = FALSE), mock_data_extract_spec(select_multiple = FALSE)),
+      list(mock_data_extract_spec(select_multiple = FALSE), mock_data_extract_spec(select_multiple = FALSE)),
+      plot_height = c(400, 100, 600),
+      plot_width = c(600, 100, 600)
+    ),
+    "teal_module"
+  )
+})
+
+testthat::test_that("tm_g_bivariate creates a module with `ui_args` that have all arguments of function", {
+  local_logger_threshold(logger::FATAL) # Suppress logger messages
+
+  mod <- tm_g_bivariate(
+    "a label",
+    mock_data_extract_spec(select_multiple = FALSE),
+    mock_data_extract_spec(select_multiple = FALSE)
+  )
+
+  testthat::expect_contains(
+    names(mod$ui_args),
+    names(formals(tm_g_bivariate))
+  )
+})
+
```

Comment:
I think you can skip this. `ui_args` contains too many elements - not sure if even half off them are really used in the ui.

```suggestion

```